### PR TITLE
Support options that is classes with props

### PR DESCRIPTION
**Pull request by mogadanez**
https://github.com/nodejs/io.js/commit/718059777c6eb41d930db787eec8748ab
35f7deb#commitcomment-11406521

### CODE REVIEW

**Comment by Fishrock123** [comment]
Should this have been PR'd to nodejs/io.js?

**Comment by mogadanez** [comment]
I think in both, same issue with current version of node

**Comment by brendanashworth** [comment]
I think you can just replace the `util._extend` entirely with the `Object.create`.

**Comment by mogadanez** [comment]
done

File: lib/_http_client.js [hunk 1]
```diff
@@ -48,7 +48,7 @@ function ClientRequest(options, cb) {
   if (util.isString(options)) {
     options = url.parse(options);
   } else {
-    options = util._extend({}, options);
+    options = Object.create(options||{});
```
Comment (brendanashworth) [inline lib/_http_client.js]: style: space between operands, `options || {}`

**Comment by jasnell** [comment]
Closing this here in favor of the corresponding nodejs/node PR. This won't be able to land here. If the PR lands in nodejs/node, we can cherry pick back to v0.12 if appropriate.

**Pull request closed without merging**